hmmlearn/_hmmc.pyx

```diff
@@ -27,13 +26,13 @@ cdef inline dtype_t _logsumexp(dtype_t[:] X) nogil:
 
 
 def _forward(int n_samples, int n_components,
-        np.ndarray[dtype_t, ndim=1] log_startprob,
-        np.ndarray[dtype_t, ndim=2] log_transmat,
-        np.ndarray[dtype_t, ndim=2] framelogprob,
-        np.ndarray[dtype_t, ndim=2] fwdlattice):
+        dtype_t[:] log_startprob,
+        dtype_t[:, :] log_transmat,
+        dtype_t[:, :] framelogprob,
+        dtype_t[:, :] fwdlattice):
 
     cdef int t, i, j
-    cdef dtype_t[:] work_buffer = np.zeros(n_components)
+    cdef dtype_t[::1] work_buffer = np.zeros(n_components)
```

Comment:
This tells cython that the buffer is contiguous. This allows cython to hardcode the strides, which saves a lookup and results in faster indexing. I should be using the `view.contiguous` specifier and have changed this.